### PR TITLE
933: Docs. Pydantic Forms from a frontend perspective.

### PR DESCRIPTION
**Pull request by DutchBen**

### CODE REVIEW

File: docs/architecture/application/forms-frontend.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+It's passed the complete property object so it can use them to adjust it's behavior.
+```
+
+-   A `POST` with the form values is made to the same `/processes/<workflow_name>` endpoint
+-   The response status code can be:
+    -   `400: Form invalid` Invalid values have been detected because the validators the backend runs have failed. An error message is shown.
+    -   `510: FormNotComplete` There is another step. This response contains another json response containing a form.
+    -   `201: Created` The workflow was initiated successfully. The response contains a workflow id and the user is redirected to the workflow detail page
+
+**Note**. For forms that have multiple steps the user input for each step is accumulated in local frontend state and posted to `/processes/<workflowname>` on each step. The endpoint will receive all available user inputs on each step and determine what other user input it till needs or if it's ready to start the workflow.
```
Comment (anieuwland) [inline docs/architecture/application/forms-frontend.md:62]: > ..it till needs..

> ..it **still** needs..

File: docs/architecture/application/forms-frontend.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+**Note 2** The Orchestrator Component library contains fields that are marked as deprecated and live in a folder named `deprectated`. These contain field types that are very specific to workflows that are in use by SURF. There are plans to remove these from the general purpose components library.
```
Comment (anieuwland) [inline docs/architecture/application/forms-frontend.md:64]: `deprectated` or `deprecated`?

**Review by Mark90 — APPROVED** [review]
Nice! Note that if you want this documentation to be published as well, that you have to add it in `mkdocs.yml`

File: docs/architecture/application/forms-frontend.md (hunk 2 above)
Comment (DutchBen) [reply docs/architecture/application/forms-frontend.md:64]: I'm pretty sure it's ``deprecated`` https://www.google.com/search?q=deprectated

Comment (DutchBen) [inline docs/architecture/application/forms-frontend.md:64]:
```suggestion
**Note 2** The Orchestrator Component library contains fields that are marked as deprecated and live in a folder named `deprecated`. These contain field types that are very specific to workflows that are in use by SURF. There are plans to remove these from the general purpose components library.
```

Comment (DutchBen) [reply docs/architecture/application/forms-frontend.md]: Fixed

**Review by anieuwland — APPROVED** [review]

File: docs/architecture/application/forms-frontend.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
+# Forms - from a frontend perspective
+
+Orchestrator Core contains a module called Pydantic Forms. Pydantic Forms allows for configuration of input forms to collect user input needed for the execution of a workflow. The module contains a frontend part that display the forms automatically and handles submission and showing validation errors. This documentation describes what happens on the frontend side of this process.
```
Comment (wouter1975) [inline docs/architecture/application/forms-frontend.md:3]: displays

File: docs/architecture/application/forms-frontend.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+## Initiating a workflow from frontend
+
+A workflow can be initiated by doing a POST call to ''/processes/<workflow_name>''
+
+The steps that happen to initiate a workflow on the frontend are:
+
+-   A `POST` request to `/processes/<workflow_name>` with an empty payload
+-   The backend determines what input values are missing and sends a response with http status code `510` and a payload containing a [JSON6Schema definition][2] describing the form to display. See [Example JSON6Schema response](#example-json6schema-response)
+-   The frontend uses the [Uniforms library][1] to parse the JSON response into a form to display
+-   The [AutofieldLoader function][3] is called for each of the form.properties in the JSON response. This functions uses the properties `type` and `format` to determine what kind of field will be displayed.
```
Comment (wouter1975) [inline docs/architecture/application/forms-frontend.md:14]: function

File: docs/architecture/application/forms-frontend.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    -   `510: FormNotComplete` There is another step. This response contains another json response containing a form.
+    -   `201: Created` The workflow was initiated successfully. The response contains a workflow id and the user is redirected to the workflow detail page
+
+**Note**. For forms that have multiple steps the user input for each step is accumulated in local frontend state and posted to `/processes/<workflowname>` on each step. The endpoint will receive all available user inputs on each step and determine what other user input it still needs or if it's ready to start the workflow.
+
+**Note 2** The Orchestrator Component library contains fields that are marked as deprecated and live in a folder named `deprecated`. These contain field types that are very specific to workflows that are in use by SURF. There are plans to remove these from the general purpose components library.
+
+**Note 3** There are plans to make it easier to extend this functionality with to add custom field types and extend the switch statement in the autoFieldFunction to include these custom `types` or `formats`
```
Comment (wouter1975) [inline docs/architecture/application/forms-frontend.md:66]: remove with?

**Review by wouter1975 — APPROVED** [review]

**Review by pboers1988 — APPROVED** [review]

**Pull request merged**